pkg/controller/atlasdeployment/searchindex.go

```diff
@@ -0,0 +1,182 @@
+		return sr.update(stateInAKO, stateInAtlas)
+	case !emptyInAtlas && emptyInAKO:
+		return sr.delete(stateInAtlas)
+	default:
+		return workflow.OK()
+	}
+}
+
+func (sr *searchIndexReconciler) idle(index *searchindex.SearchIndex) workflow.Result {
+	sr.ctx.Log.Debugf("[idle] index '%s'", index.Name)
+	sr.ctx.EnsureStatusOption(status.AtlasDeploymentSetSearchIndexStatus(status.NewDeploymentSearchIndexStatus(
+		status.SearchIndexStatusReady,
+		status.WithID(index.GetID()),
+		status.WithName(index.Name),
+		status.WithMsg(index.GetStatus()))))
+	ok := workflow.OK()
+	sr.ctx.SetConditionFromResult(status.SearchIndexesReadyType, ok)
+	return ok
+}
+
+// Never set the ID (status.WithID()) on terminate. It may be empty and the AKO will lose track on this index
+func (sr *searchIndexReconciler) terminate(index *searchindex.SearchIndex, err error) workflow.Result {
+	msg := fmt.Errorf("error with processing index %s. err: %w", index.Name, err)
+	sr.ctx.Log.Debug(msg)
+	sr.ctx.EnsureStatusOption(status.AtlasDeploymentSetSearchIndexStatus(status.NewDeploymentSearchIndexStatus(
+		status.SearchIndexStatusError,
+		status.WithMsg(msg.Error()),
+		status.WithName(index.Name),
+	)))
+	terminate := workflow.Terminate(status.SearchIndexStatusError, msg.Error())
+	sr.ctx.SetConditionFromResult(status.SearchIndexesReadyType, terminate)
+	return terminate
+}
+
+func (sr *searchIndexReconciler) create(index *searchindex.SearchIndex) workflow.Result {
+	sr.ctx.Log.Debugf("[creating] index %s", index.Name)
+	defer sr.ctx.Log.Debugf("[creation finished] for index %s", index.Name)
+	atlasIdx, err := index.ToAtlas()
+	if err != nil {
+		return sr.terminate(index, err)
+	}
+	respIdx, resp, err := sr.ctx.SdkClient.AtlasSearchApi.CreateAtlasSearchIndex(sr.ctx.Context,
+		sr.projectID, sr.deployment.GetDeploymentName(), atlasIdx).Execute()
+	if err != nil || resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusOK {
+		return sr.terminate(index, fmt.Errorf("failed to create index: %w, status: %d", err, resp.StatusCode))
+	}
+	if respIdx == nil {
+		return sr.terminate(index, fmt.Errorf("returned an empty index as a result of creation"))
+	}
+	akoIdx, err := searchindex.NewSearchIndexFromAtlas(*respIdx)
+	if err != nil {
+		return sr.terminate(index, fmt.Errorf("unable to convert index to AKO: %w", err))
+	}
+	return sr.progress(akoIdx)
+}
+
+func (sr *searchIndexReconciler) progress(index *searchindex.SearchIndex) workflow.Result {
+	sr.ctx.Log.Debugf("index %s is progress: %s", index.Name, index.GetStatus())
+	sr.ctx.EnsureStatusOption(status.AtlasDeploymentSetSearchIndexStatus(
+		status.NewDeploymentSearchIndexStatus(status.SearchIndexStatusInProgress,
+			status.WithMsg(pointer.GetOrDefault(index.Status, "")),
+			status.WithID(index.GetID()),
+			status.WithName(index.Name))))
+	inProgress := workflow.InProgress(status.SearchIndexStatusInProgress, index.GetStatus())
+	sr.ctx.SetConditionFromResult(status.SearchIndexesReadyType, inProgress)
```

Comment:
this yields:
```
    - lastTransitionTime: "2024-05-03T05:43:37Z"
      message: IN_PROGRESS
      reason: InProgress
      status: "False"
      type: AtlasSearchIndexes
```

which is not very informative. I would expect `message` to be more expanded to have more context, i.e.:
```
    - lastTransitionTime: "2024-05-03T05:43:37Z"
      message: Atlas Search Index status: IN_PROGRESS
      reason: InProgress
      status: "False"
      type: AtlasSearchIndexes
```

Reply:
`AtlasSearchIndexes` renamed to `AtlasSearchIndexesReady`. In case index is in progress, the message now contains the "Atlas search index status"